services/workflows-service/src/note/note.model.ts

```diff
@@ -0,0 +1,60 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { EntityType, Noteable } from '@prisma/client';
+import { IsString } from 'class-validator';
+
+export class NoteModel {
+  @ApiProperty({
+    required: true,
+    type: String,
+  })
+  @IsString()
+  id!: string;
+
+  @ApiProperty({
+    required: true,
+    type: String,
+  })
+  @IsString()
+  entityId!: string;
+
+  @ApiProperty({
+    required: true,
+    enum: ['Business', 'EndUser'],
+  })
+  @IsString()
+  entityType!: EntityType;
+
+  @ApiProperty({
+    type: String,
+  })
+  @IsString()
+  noteableId!: string;
+
+  @ApiProperty({
+    required: true,
+    enum: ['Workflow', 'Report', 'Alert'],
+  })
+  @IsString()
+  noteableType!: Noteable;
+
+  @ApiProperty({
+    required: true,
+    type: String,
+  })
+  @IsString()
+  content!: string;
+
+  @ApiProperty({
+    required: true,
+    type: Object,
+  })
+  @IsString()
+  parentNote!: NoteModel | null;
```

Comment:
IsString is a mistake? Looks as though NoteModel is supposed to be passed into ApiProperty somehow and null not to be only on the type level.